### PR TITLE
Indirect Evaporative Cooler Research Special does not always set the secondary air outlet node conditions

### LINKED ISSUE #6211: EvaporativeCooler:Indirect:ResearchSpecial does not always set the secondary air outlet node conditions
Issue overview
--------------
EvaporativeCooler:Indirect:ResearchSpecial only sets the secondary air outlet node conditions when these four fields have inputs:

```
EvaporativeCooler:Indirect:ResearchSpecial,

    0.75,                    !- Cooler Drybulb Design Effectiveness

    -99,                     !- Evaporative Operation Minimum Limit Secondary Air Drybulb Temperature
    99,                      !- Evaporative Operation Maximum Limit Outdoor Wetbulb Temperature
    99;                      !- Dry Operation Maximum Limit Outdoor Drybulb Temperature

```
If any one of these fields is blank, then `EvapCoolerOperationControlFlag` is set to false [here](https://github.com/NREL/EnergyPlus/blob/9a786ca0a17b0eb97c4641f31663b44367a3d54f/src/EnergyPlus/EvaporativeCoolers.cc#L645-L652)

And if `EvapCoolerOperationControlFlag` is false, then the secondary outlet node conditions are skipped [here](https://github.com/NREL/EnergyPlus/blob/9a786ca0a17b0eb97c4641f31663b44367a3d54f/src/EnergyPlus/EvaporativeCoolers.cc#L3212-L3217).  Not sure why this should be skipped here.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus v8.7
 - Helpdesk ticket number 12149

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
This resolves #6211. The advanced control feature may turn-off the evaporative cooler due to incorrect logic. This is addressed now. IO reference documentation was update to explain the advanced control feature requirements. Diffs are expected/justified based on this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - [ ] Required documentation updates?

### CODE REVIEW

**Comment by Nigusse** [comment]
Addressed the two problems described in issue #6211:
(1) corrected logic that turns-off the evap cooler when the primary and secondary air inlet temperatures are equal, (2) updated IO reference documentation and added notes to idd describing requirements for advanced control feature proper operation. Tested the defect file, now runs as expected. Maximum iteration error message goes away if you add effectiveness modifier curves. This is ready for review.

File: src/EnergyPlus/EvaporativeCoolers.cc [hunk 1]
```diff
@@ -2880,6 +2858,56 @@ namespace EvaporativeCoolers {
...
+        Real64 SysTempSetPoint;     // evaporative cooler outlet setpoint temperature, drybulb
+        int OperatingMode;          // current operating mode of indrect evaporative cooler
+
+        InletDryBulbTempPri = EvapCond(EvapCoolNum).InletTemp;
+        SysTempSetPoint = EvapCond(EvapCoolNum).DesiredOutletTemp;
+
+        // Now determine the operating modes of indirect evaporative cooler research special. There are five allowed operating modes
+        if ((InletDryBulbTempPri <= SysTempSetPoint) ||
+            (InletDryBulbTempPri > EvapCond(EvapCoolNum).MaxOATDBEvapCooler && InletWetBulbTempSec > EvapCond(EvapCoolNum).MaxOATWBEvapCooler) ||
+            (InletDryBulbTempPri < InletDryBulbTempSec)) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/EvaporativeCoolers.cc:2891]: OK, so here's the fix, good.

File: src/EnergyPlus/EvaporativeCoolers.cc [hunk 2]
```diff
@@ -2880,6 +2858,56 @@ namespace EvaporativeCoolers {
...
+        } else if ((InletDryBulbTempSec >= EvapCond(EvapCoolNum).MinOATDBEvapCooler &&
+                    InletDryBulbTempSec < EvapCond(EvapCoolNum).MaxOATDBEvapCooler &&
+                    InletWetBulbTempSec < EvapCond(EvapCoolNum).MaxOATWBEvapCooler && SysTempSetPoint < TdbOutSysDryMin &&
+                    TdbOutSysWetMin < SysTempSetPoint)) {
+            OperatingMode = DryWetModulated; // modulated in dry and wet mode, and the lower total power will be used
+        } else {
+            OperatingMode = None; // this condition should not happen unless the bounds do not cover all combinations possible
+        }
+        return OperatingMode;
+    }
```
Comment (Myoldmopar) [inline src/EnergyPlus/EvaporativeCoolers.cc:2913]: I like that you extracted this into a proper worker function with a singular purpose.  Easy to test and maintain going forward.

File: tst/EnergyPlus/unit/EvaporativeCoolers.unit.cc [hunk 3]
```diff
@@ -761,4 +761,47 @@ TEST_F(EnergyPlusFixture, DefaultAutosizeDirEvapCoolerTest)
...
+    EXPECT_EQ(Result_WetFullOperatingMode, EvaporativeCoolers::WetFull);
+    // get outlet temperature in full wet operating mode
+    EvaporativeCoolers::CalcIndirectRDDEvapCoolerOutletTemp(EvapCoolNum,
+                                                            Result_WetFullOperatingMode,
+                                                            thisEvapCooler.SecInletMassFlowRate,
+                                                            thisEvapCooler.SecInletTemp,
+                                                            thisEvapCooler.SecInletWetBulbTemp,
+                                                            thisEvapCooler.SecInletHumRat);
+    // test outlet temperature in full wet operating mode
+    EXPECT_NEAR(22.03, thisEvapCooler.OutletTemp, 0.001);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/EvaporativeCoolers.unit.cc:804]: :+1:

**Review by Myoldmopar — APPROVED** [review]
Everything here looks good to go.

**Comment by Myoldmopar** [comment]
Merged locally to resolve conflicts, everything seems good.  Merging now.

**Pull request merged**